### DISCUSSION TITLE
SpanKind for root spans in Java thick client

### DISCUSSION THREAD

**Discussion by k-rloss** [open-telemetry/opentelemetry-java#6598] [Q&A]
Hi
I'm instrumenting a Java client-server application (Java Swing application that sends requests to a couple of REST endpoints), and I am interested in capturing spans that closely track user interactions at the UI level. For example, if a user hits a button, and it takes the application 2 seconds to provide the result back to the user (after having made http calls to the backend), I'd like to have a **root span** that covers those 2 seconds.

I plan to explicitly start those root spans as follows:

`Span span = tracer.spanBuilder("UserOperation1").setSpanKind(SpanKind.SERVER).startSpan();`

In this scenario, is setting the span kind to SERVER the recommended approach ?  (Since those root spans are the "entry points" to the client layer)

One of my co-workers was suggesting that SERVER spans should not be used because these spans are being captured in the client layer.

Thanks

**Comment by jkwatson** [comment]
I think I'd just use INTERNAL in this case. There isn't really a specific span kind that maps to this usage.
